portal/templatetags/app_tags.py

```diff
@@ -87,12 +87,7 @@ def is_logged_in_as_teacher(u):
 @register.filter(name='has_teacher_finished_onboarding')
 def has_teacher_finished_onboarding(u):
     teacher = u.userprofile.teacher
-    teacher_has_class = teacher.has_school() and teacher.has_class()
-    if teacher_has_class:
-        class_ = teacher.class_()
-        return is_logged_in_as_teacher(u) and teacher_has_class and class_.has_students()
-    else:
-        return False
+    return is_logged_in_as_teacher(u) and teacher.has_school() and teacher.class_() is not None and teacher.class_().has_students()
```

Comment:
please assign teacher.class_ to a variable so it doesn't compute it twice in case of success!